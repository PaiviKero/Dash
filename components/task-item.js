import React, { useState } from "react";

import { Input } from "./styles";
import { strings } from "resources/strings";

export const TaskItem = ({ task = {} }) => {
  const [name, setName] = useState(task.name);

  // TODO: saving task somewhere ????
  return (
    <>
      <Input
        label={strings.task.name}
        value={name}
        // TODO: test changing text
        onChangeText={(text) => setName(text)}
      />
    </>
  );
};
