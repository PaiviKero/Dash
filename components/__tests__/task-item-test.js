import "react-native";
import React from "react";
import { TaskItem } from "../task-item";

import renderer from "react-test-renderer";

it("TaskItem renders correctly", () => {
  const tree = renderer.create(<TaskItem />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("TaskItem name property renders correctly", () => {
  const otherTask = { name: "Other task" };
  const tree = renderer.create(<TaskItem task={otherTask} />).toJSON();
  expect(tree).toMatchSnapshot();
});
