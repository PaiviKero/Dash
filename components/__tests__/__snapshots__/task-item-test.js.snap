// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskItem name property renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(231, 231, 231)",
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
      },
      Object {
        "width": 300,
      },
    ]
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.26)",
        "bottom": 0,
        "height": 2,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "transform": Array [
          Object {
            "scaleY": 0.5,
          },
        ],
        "zIndex": 1,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "paddingBottom": 0,
          "paddingTop": 0,
        },
        Object {
          "minHeight": 60,
        },
      ]
    }
  >
    <View
      collapsable={false}
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 3,
            },
          ],
          "zIndex": 3,
        }
      }
    >
      <Text
        collapsable={false}
        maxFontSizeMultiplier={1.5}
        numberOfLines={1}
        onLayout={[Function]}
        style={
          Object {
            "color": "#6200ee",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
            "left": 0,
            "opacity": 0,
            "paddingLeft": 12,
            "paddingRight": 12,
            "position": "absolute",
            "textAlign": "left",
            "top": 30,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": -12,
              },
              Object {
                "scale": 0.75,
              },
            ],
            "writingDirection": "ltr",
          }
        }
      >
        Task Name
      </Text>
      <Text
        collapsable={false}
        maxFontSizeMultiplier={1.5}
        numberOfLines={1}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
            "left": 0,
            "opacity": 0,
            "paddingLeft": 12,
            "paddingRight": 12,
            "position": "absolute",
            "textAlign": "left",
            "top": 30,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": -12,
              },
              Object {
                "scale": 0.75,
              },
            ],
            "writingDirection": "ltr",
          }
        }
      >
        Task Name
      </Text>
    </View>
    <TextInput
      editable={true}
      maxFontSizeMultiplier={1.5}
      multiline={false}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder=""
      placeholderTextColor="rgba(0, 0, 0, 0.54)"
      selectionColor="#6200ee"
      style={
        Array [
          Object {
            "flexGrow": 1,
            "margin": 0,
          },
          Object {
            "paddingLeft": 12,
            "paddingRight": 12,
          },
          Object {
            "height": 60,
          },
          Object {
            "paddingBottom": 4,
            "paddingTop": 24,
          },
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 16,
            "fontVariant": undefined,
            "fontWeight": undefined,
            "textAlign": "left",
            "textAlignVertical": "center",
          },
          false,
          Array [
            Object {},
          ],
        ]
      }
      underlineColorAndroid="transparent"
      value="Other task"
    />
  </View>
</View>
`;

exports[`TaskItem renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(231, 231, 231)",
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
      },
      Object {
        "width": 300,
      },
    ]
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.26)",
        "bottom": 0,
        "height": 2,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "transform": Array [
          Object {
            "scaleY": 0.5,
          },
        ],
        "zIndex": 1,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "paddingBottom": 0,
          "paddingTop": 0,
        },
        Object {
          "minHeight": 60,
        },
      ]
    }
  >
    <View
      collapsable={false}
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
          "zIndex": 3,
        }
      }
    >
      <Text
        collapsable={false}
        maxFontSizeMultiplier={1.5}
        numberOfLines={1}
        onLayout={[Function]}
        style={
          Object {
            "color": "#6200ee",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
            "left": 0,
            "opacity": 0,
            "paddingLeft": 12,
            "paddingRight": 12,
            "position": "absolute",
            "textAlign": "left",
            "top": 30,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
              Object {
                "scale": 1,
              },
            ],
            "writingDirection": "ltr",
          }
        }
      >
        Task Name
      </Text>
      <Text
        collapsable={false}
        maxFontSizeMultiplier={1.5}
        numberOfLines={1}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
            "left": 0,
            "opacity": 0,
            "paddingLeft": 12,
            "paddingRight": 12,
            "position": "absolute",
            "textAlign": "left",
            "top": 30,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
              Object {
                "scale": 1,
              },
            ],
            "writingDirection": "ltr",
          }
        }
      >
        Task Name
      </Text>
    </View>
    <TextInput
      editable={true}
      maxFontSizeMultiplier={1.5}
      multiline={false}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder=""
      placeholderTextColor="rgba(0, 0, 0, 0.54)"
      selectionColor="#6200ee"
      style={
        Array [
          Object {
            "flexGrow": 1,
            "margin": 0,
          },
          Object {
            "paddingLeft": 12,
            "paddingRight": 12,
          },
          Object {
            "height": 60,
          },
          Object {
            "paddingBottom": 4,
            "paddingTop": 24,
          },
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 16,
            "fontVariant": undefined,
            "fontWeight": undefined,
            "textAlign": "left",
            "textAlignVertical": "center",
          },
          false,
          Array [
            Object {},
          ],
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;
